Code/GraphMol/MolDraw2D/DrawMol.cpp

```diff
@@ -3562,6 +3562,19 @@ DrawColour DrawMol::getColour(int atom_idx) const {
         retval = *highCol;
       }
     }
+  } else {
+    // There's going to be a colour behind the atom, so if the
+    // atom has a symbol, it should be the same colour as carbon.  This
+    // function should only be called if there is an atom symbol.
+    if (highlightedAtom) {
+      if (drawOptions_.atomColourPalette.find(6) !=
+          drawOptions_.atomColourPalette.end()) {
+        retval = drawOptions_.atomColourPalette.find(6)->second;
+      } else {
+        // Use the default if no carbon.
+        retval = drawOptions_.atomColourPalette.find(-1)->second;
```

Comment:
This would be cleaner (and safer) as:
```suggestion
      if (drawOptions_.atomColourPalette.find(6) !=
          drawOptions_.atomColourPalette.end()) {
        retval = drawOptions_.atomColourPalette.at(6);
      } else {
        // Use the default if no carbon.
        retval = drawOptions_.atomColourPalette.at(-1);
```